### PR TITLE
Added possibility to lint tests

### PR DESCRIPTION
**Pull request by gharlan**
Let us look how many test cases are invalid.

### CODE REVIEW

**Comment by keradus** [comment]
I was thinking about that too ;)
Please, check the `$input` too.

**Comment by gharlan** [comment]
PHP version | Invalid test cases
--- | ---
PHP 5.3 | 219
PHP 5.4 | 81
PHP 5.5 | 78
PHP 5.6 | 78
PHP 7.0 | 82

**Comment by GrahamCampbell** [comment]
LOL

**Comment by SpacePossum** [comment]
@gharlan could you add messages to the test so it is a bit more clear to the developers what failed?
If you agree with https://github.com/SpacePossum/PHP-CS-Fixer/commit/6c199933b39f87a74a97cab2ef5c0b3ef6cb0758 could you pick it on your branch? I'll drop the PR (https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/1545) and move on with the next bunch of tests that needs fixing :)
(en please rebase ;) )

**Comment by gharlan** [comment]
@SpacePossum I will rebase when your PR is merged.

**Comment by keradus** [comment]
good plan

**Comment by SpacePossum** [comment]
:+1:  sounds good to me :)

**Comment by gharlan** [comment]
Update:

PHP version | Invalid test cases
--- | ---
PHP 5.3 | 204
PHP 5.4 | 47
PHP 5.5 | 41
PHP 5.6 | 41
PHP 7.0 | 48

**Comment by gharlan** [comment]
So, what is to do here?
Only linting on highest php version?

**Comment by keradus** [comment]
OK, please lint only there, and when we handle it we will think about next step

**Comment by SpacePossum** [comment]
shoud(n't) we lint the intergration test input as well?
(https://github.com/FriendsOfPHP/PHP-CS-Fixer/blob/1.12/Symfony/CS/Tests/AbstractIntegrationTest.php#L147)

**Comment by GrahamCampbell** [comment]
> shoud(n't) we lint the intergration test input as well?

:+1:

**Comment by gharlan** [comment]
> OK, please lint only there

How should I do this? hard coded check for php 7? environment variable? or..?

**Comment by keradus** [comment]
What do you think about using an env for PHPUnit?

**Comment by gharlan** [comment]
done. that way?

**Comment by keradus** [comment]
Yeah, but please also add it to `phpunit.xml.dist` file with default value. Just to have listed env variables that are used inside phpunit tests.

**Comment by gharlan** [comment]
> Yeah, but please also add it to phpunit.xml.dist file with default value.

done

> shoud(n't) we lint the intergration test input as well?

done

**Comment by SpacePossum** [comment]
thanks! :+1:

**Comment by gharlan** [comment]
I have a general question: When should I add my `@author` tag to a modified class and when not to?
Is there any rule?

**Comment by keradus** [comment]
When you think that your contribution to the class is significant

**Comment by gharlan** [comment]
At the moment we are linting: 

* input and expected of test cases
* input of integration tests

IMO we should either add the expected part of integration tests, or remove the expected part from the other test cases (when input is present).

By the way: I don't like the difference between integration tests and other test cases regarding input/expected. IMO we should change the integrations tests, so that the expected part is mandatory, and the input is optional. And we should swap input and expected (first expected, then input).

**Comment by keradus** [comment]
> IMO we should either add the expected part of integration tests, or remove the expected part from the other test cases (when input is present).

We should check both.



> IMO we should change the integrations tests, so that the expected part is mandatory, and the input is optional. And we should swap input and expected (first expected, then input).

big :+1: for that, I actually trap into that few times!
Could you take care of it?

**Comment by gharlan** [comment]
> Could you take care of it?

which branch?

**Comment by keradus** [comment]
`AbstractIntegrationTest` is internal on 1.x so you could change it on 1.x

**Comment by SpacePossum** [comment]
>And we should swap input and expected (first expected, then input).

does that mean swapping the parts in the fixtures file (i.e. `.test`) as well?

**Comment by SpacePossum** [comment]
>input of integration tests

maybe lint the output as well?

**Comment by gharlan** [comment]
yes

**Comment by gharlan** [comment]
As #1594 is merged into 1.12, should I open a new pr for 1.12?

File: .travis.yml [hunk 1]
```diff
@@ -12,6 +12,7 @@ matrix:
         - php: 5.6
           env: SYMFONY_VERSION="~3.0@dev" COMPOSER_FLAGS="--prefer-stable"
         - php: 7.0
+          env: LINT_TEST_CASES=1
```
Comment (keradus) [inline .travis.yml]: I think that for now we could remove that line, merge the branch (so no changes for Travis) and then, when we get all test base valid, we will add it again. What do you think?

Comment (gharlan) [inline .travis.yml]: good plan :+1: 
line removed.

**Comment by keradus** [comment]
Thank you @gharlan.

**Comment by gharlan** [comment]
@keradus there is a conflict with #1605.

**Comment by keradus** [comment]
Could you ref it?

**Pull request closed without merging**